--- hyperscan-5.0.0/CMakeLists.txt	2018-07-09 09:45:44.000000000 -0700
+++ hyperscan-5.0.0-patched/CMakeLists.txt	2018-12-30 12:34:10.752774012 -0800
@@ -231,8 +231,8 @@
     endif(OPTIMISE)
 
     # set compiler flags - more are tested and added later
-    set(EXTRA_C_FLAGS "${OPT_C_FLAG} -std=c99 -Wall -Wextra -Wshadow -Wcast-qual -fno-strict-aliasing")
-    set(EXTRA_CXX_FLAGS "${OPT_CXX_FLAG} -std=c++11 -Wall -Wextra -Wshadow -Wswitch -Wreturn-type -Wcast-qual -Wno-deprecated -Wnon-virtual-dtor -fno-strict-aliasing")
+    set(EXTRA_C_FLAGS "${OPT_C_FLAG} -std=c99 -Wall -Wextra -Wshadow -Wcast-qual -fno-strict-aliasing -g3")
+    set(EXTRA_CXX_FLAGS "${OPT_CXX_FLAG} -std=c++11 -Wall -Wextra -Wshadow -Wswitch -Wreturn-type -Wcast-qual -Wno-deprecated -Wnon-virtual-dtor -fno-strict-aliasing -g3")
 
     if (NOT RELEASE_BUILD)
         # -Werror is most useful during development, don't potentially break
