diff --git a/lib/B/C.pm b/lib/B/C.pm
index ee1f8f72..97454cad 100644
--- a/lib/B/C.pm
+++ b/lib/B/C.pm
@@ -7948,6 +7948,44 @@ EOT1
 }
 
 sub output_init {
+   print <<'EOT';
+#if defined(USE_ITHREADS)
+    {
+        /* added by OpenResty Inc. */
+        unsigned i = 0;
+        for (i = 0; i < sizeof(sv_list) / sizeof(sv_list[0]); i++) {
+            SV *sv = &sv_list[i];
+            if (SvREFCNT(sv) == 1) {
+                char *pvx;
+                XPV *xpv;
+                size_t len;
+
+                pvx = SvPVX_mutable(sv);
+                if (!pvx) {
+                    continue;
+                }
+                xpv = (XPV *) SvANY(sv);
+                if (!xpv
+                    || xpv < xpv_list
+                    || (char *) xpv + sizeof(XPV) > (char *) xpv_list + sizeof(xpv_list))
+                {
+                    continue;
+                }
+
+                len = SvLEN(sv);
+                if (len) {
+                    char *p;
+                    p = malloc(len + 1);
+                    assert(p != NULL);
+                    memcpy(p, pvx, len + 1);
+                    SvPVX(sv) = p;
+                }
+            }
+        }
+    }
+#endif
+EOT
+
   print <<'EOT';
   /* our special compiled init */
     perl_init(aTHX);
