From patchwork Sun Nov  5 17:09:51 2017
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
Subject: [v3,2/2] Input: synaptics - Lenovo X1 Carbon 5 should use SMBUS/RMI
From: Yiannis Marangos <yiannis.marangos@gmail.com>
X-Patchwork-Id: 10042387
Message-Id: <20171105170951.20261-3-yiannis.marangos@gmail.com>
To: linux-input@vger.kernel.org, dmitry.torokhov@gmail.com
Cc: Yiannis Marangos <yiannis.marangos@gmail.com>
Date: Sun,  5 Nov 2017 19:09:51 +0200

Signed-off-by: Yiannis Marangos <yiannis.marangos@gmail.com>
---
 drivers/input/mouse/synaptics.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/input/mouse/synaptics.c b/drivers/input/mouse/synaptics.c
index ee5466a374bf..b3c683a84d3f 100644
--- a/drivers/input/mouse/synaptics.c
+++ b/drivers/input/mouse/synaptics.c
@@ -172,6 +172,7 @@ static const char * const smbus_pnp_ids[] = {
 	"LEN0048", /* X1 Carbon 3 */
 	"LEN0046", /* X250 */
 	"LEN004a", /* W541 */
+	"LEN0073", /* X1 Carbon 5 */
 	"LEN200f", /* T450s */
 	NULL
 };
